File: Level1/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int n) {

    int answer = 0;
    vector<int> vect;

    // 3진법으로 변환 & 앞뒤 반전
    while (n)
    {
        vect.push_back(n % 3);
        n /= 3;
    }

    // 10진법으로 표현
    for (int i = 0; i < vect.size(); i++)
        answer += vect[i] * pow(3, vect.size() - i - 1);

    return answer;
}

File: Python/Level1/3진법 뒤집기.py
def solution(n):
    n3, i = 0, 0
    while n:
        n3 = n3 * 10 + (n % 3)
        n //= 3
    while n3:
        n += pow(3, i) * (n3 % 10)
        n3 //= 10
        i += 1
    return n
